src/controllers/VueController.php

```diff
@@ -117,12 +77,16 @@ protected static function convertCurrency(float $amount, string $currency, bool
      */
     private function fetchOrders($id = null) : array
     {
-        $single = $_GET['purchasableId'] ?? $id;
-        $start  = $this->start_date ?: '1970-01-01 00:00:00';
-        $end    = $this->end_date ?: gmdate('Y-m-d 23:59:59');
-        $orders = Order::find()->dateOrdered(['and', ">= {$start}", "< {$end}"]);
-        $url    = $_SERVER['REQUEST_URI'];
-        $path   = basename(parse_url($url, PHP_URL_PATH));
+        $single   = $_GET['purchasableId'] ?? $id;
+        $start    = $this->start_date ? \DateTime::createFromFormat('Y-m-d H:i:s', $this->start_date) : '1970-01-01 00:00:00';
```

Comment:
I think that previous and current ranges should be kept separate. Combining the 2 ranges into a single resultset can be confusing/complex to work with on the client, and makes maintenance harder on both the client and the API.